test/mpi/configure.ac

```diff
@@ -358,37 +355,23 @@ if test -n "$F90" -o -n "$F90FLAGS" ; then
     AC_MSG_ERROR([F90 and F90FLAGS are replaced by FC and FCFLAGS respectively in this configure, please unset F90/F90FLAGS and set FC/FCFLAGS instead and rerun configure again.])
 fi
 
-# ------------------------------------------------------------------------
-dnl use AC_ARG_VAR to mark FROM_MPICH as "precious" to autoconf so that
-dnl automatic re-runs of config.status preserve its value correctly
-AC_ARG_VAR([FROM_MPICH],[should be set to "yes" if this configure script is being invoked by the main MPICH configure])
-AC_ARG_VAR([MPICH_THREAD_LEVEL],
-           [the MPI thread level supported by the enclosing MPICH build (when built within MPICH)])
-# ------------------------------------------------------------------------
-
+# check whether to enable thread tests
+#
 if test "$enable_threads" = "yes" ; then
     enable_threads=multiple
 elif test "$enable_threads" = "no" ; then
     enable_threads=single
 elif test "$enable_threads" = default ; then
-    if test -n "$MPICH_THREAD_LEVEL" ; then
-        case $MPICH_THREAD_LEVEL in
-            MPI_THREAD_MULTIPLE)
-            enable_threads=multiple
-            ;;
-            MPI_THREAD_SERIALIZED)
-            enable_threads=serialized
-            ;;
-            MPI_THREAD_FUNNELED)
-            enable_threads=funneled
-            ;;
-            MPI_THREAD_SINGLE)
-            enable_threads=single
-            ;;
-        esac
-    else
-        enable_threads=funneled
-    fi
+    # I believe AC_TRY_RUN here is OK. The worst case (for cross compile) will result in
+    # multi-thread tests disabled. By pass by explicitly configure with --enable-threads=xxx
+    AC_TRY_RUN(AC_LANG_PROGRAM([#include "mpi.h"],[
+            int provided = MPI_THREAD_SINGLE;
+            MPI_Init_thread(0,0,MPI_THREAD_MULTIPLE,&provided);
+            MPI_Finalize();
+            if (provided != MPI_THREAD_MULTIPLE) exit(1);
+        ]),
+        [enable_threads=multiple], [enable_threads=single], [enable_threads=single]
+    )
```

Comment:
I think I'd rather just default to `multiple` and let the user opt-out if needed.

Reply:
Sounds good. That'll simplify it.

Reply:
https://github.com/pmodels/mpich/pull/5067/commits/0b039b79243421810341dc589fa5e214ee0f250d